### PR TITLE
Remove GenericTimeSeries.index

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunpy/issues/6528

Things will break.

- [x] Fix that example.

### CODE REVIEW

File: sunpy/timeseries/timeseriesbase.py [hunk 1]
```diff
@@ -313,7 +304,7 @@ def _repr_html_(self):
         for i in range(len(self.columns)):
             if len(self.columns) == 1:
                 axs.plot(
-                    dat.index,
+                    da.indext,
```
Comment (wtbarnes) [inline sunpy/timeseries/timeseriesbase.py:307]: Is this a typo?

File: changelog/6327.deprecation.rst [hunk 2]
```diff
@@ -1,3 +1,3 @@
-Using `sunpy.timeseries.GenericTimeSeries.index` is deprecated.
```
Comment (wtbarnes) [inline changelog/6327.deprecation.rst:1]: The filename is wrong here.

**Review by wtbarnes — COMMENTED** [review]
There were some problems with the doc build, maybe your most recent commits fix those? I think they were mostly just broken references.

Are there other things you're expecting to break besides that example?

File: sunpy/timeseries/timeseriesbase.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/timeseries/timeseriesbase.py:307]: Damn it, how did I miss that.

File: changelog/6327.deprecation.rst (hunk 2 above)
Comment (nabobalis) [reply changelog/6327.deprecation.rst:1]: This is an old changelog we didnt remove?!

Comment (nabobalis) [inline changelog/6327.deprecation.rst:1]: Why is this changelog still on this branch?

**Comment by nabobalis** [comment]
> There were some problems with the doc build, maybe your most recent commits fix those? I think they were mostly just broken references.

Those refs should be fixed now.
I am confused about that double changelog.

> Are there other things you're expecting to break besides that example?

No.

**Comment by nabobalis** [comment]
I don't think we did deprecate this in 4.0, I think the only thing that should last here is the example.

**Comment by dstansby** [comment]
My bad, we only deprecated this in 4.1. I think this PR can be closed then? Sorry again...

**Comment by nabobalis** [comment]
The example!

**Pull request closed without merging**